packages/rich-text/src/plugins/Hr/index.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as Slate from 'slate-react';
+import { css, cx } from 'emotion';
+import tokens from '@contentful/forma-36-tokens';
+import { EditorToolbarButton } from '@contentful/forma-36-react-components';
+import { Transforms, Editor } from 'slate';
+import { BLOCKS } from '@contentful/rich-text-types';
+import { useCustomEditor } from '../../hooks/useCustomEditor';
+import { CustomElement, CustomEditor } from 'types';
+
+const styles = {
+  container: css`
+    margin: 0 0 ${tokens.spacingL};
+  `,
+  hr: css`
+    margin: 0;
+    height: ${tokens.spacingM};
+    background: transparent;
+    position: relative;
+    border: 0;
+    user-select: none;
+    &:hover {
+      cursor: pointer;
+    }
+    &::after {
+      content: '';
+      position: absolute;
+      width: 100%;
+      height: 1px;
+      background: ${tokens.colorElementMid};
+      top: 50%;
+    }
+  `,
+  hrSelected: css`
+    &::after {
+      background: ${tokens.colorPrimary};
+      -webkit-box-shadow: 0px 0px 5px ${tokens.colorPrimary};
+      box-shadow: 0px 0px 5px ${tokens.colorPrimary};
+    }
+  `,
+};
+
+interface ToolbarHrButtonProps {
+  isDisabled?: boolean;
+}
+
+export function withHrEvents(editor: CustomEditor, event: KeyboardEvent) {
+  if (!editor.selection) return;
+
+  const [currentFragment] = Editor.fragment(editor, editor.selection.focus.path) as CustomElement[];
+  const isEnter = event.keyCode === 13;
```

Comment:
You can use `event.key === 'Enter'`, which is clearer than the key code.

We should probably also check that there are no modifier keys on the event.

Reply:
Yeah, in this case, the name of the variable already tells us this an `Enter`.